app/services/candidate_interface/continuous_applications/application_choice_submission.rb

```diff
@@ -6,13 +6,19 @@ class ApplicationChoiceSubmission
 
       delegate :application_form, to: :application_choice
       validates :application_choice,
-                cycle_verification: true,
-                your_details_completion: true,
-                submission_availability: true,
-                open_for_applications: true,
-                course_availability: true,
-                can_add_more_choices: true,
-                immigration_status: true
+                immigration_status: true,
+                applications_closed: { if: :validate_choice? },
+                course_unavailable: { if: :validate_choice? },
+                incomplete_details: { if: :validate_choice? },
+                incomplete_primary_course_details: { if: :validate_choice? },
+                incomplete_including_primary_course_details: { if: :validate_choice? },
+                can_add_more_choices: true
+
+    private
+
+      def validate_choice?
+        errors.exclude?(:application_choice)
```

Comment:
An interesting approach, so the business object only will hold one of the errors always? 🤔

Reply:
Yeah, I'm confident that the validations are called in the order they're defined.